manager.go

```diff
@@ -1318,3 +1320,31 @@ func (m *jobManager) finishJob(name string) {
 
 	delete(m.muJob.running, name)
 }
+
+type requestsClient struct {
+	lock     *sync.Mutex
+	requests *map[string]*JobRequest
+}
+
+func (m *jobManager) requestsClient() *requestsClient {
+	return &requestsClient{lock: &m.requestsLock, requests: &m.requests}
```

Comment:
jobManager should have a map by build cluster to client:

```
type ClientForCluster struct {
   CoreClient *coreclient.Interface
   ProjectsClient *projectclient.Interface
   ...
}

type jobManager struct {
  ... 
  clientLock sync.Mutex
  clients map[string]ClientsForCluster
}

func (m *jobManager) Clients(cluster string) {
  m.clientLock.Lock()
  defer m.clientLock.Unlock()
  clients, ok := m.clients[cluster]
  if ok {
    return clients
  }
  ... init clients from the configs
  m.clients[cluster] = clients
  return clients
}

```

Then you have to pass the cluster name around in the job request (and serialize to the annotations and back)